File: Go/K02.go
package main

import (
	"fmt"
	"strconv"
)

type K02 struct {

}

func (k02 K02) question1() {
	seisuu := 3
	var jissuu float32 = 2.6
	moji := "A"

	fmt.Println(fmt.Sprintf("変数seisuuの値は%d", seisuu))
	fmt.Println(fmt.Sprintf("実数jissuuの値は%f", jissuu))
	fmt.Println(fmt.Sprintf("文字mojiの値は%s", moji))
}

func (k02 K02) question2() {
	fmt.Print("一つ目の整数は？")
	var input string
	_, _ = fmt.Scanln(&input)
	var a, _ = strconv.Atoi(input)

	fmt.Print("二つ目の整数は？")
	_, _ = fmt.Scanln(&input)
	var b, _ = strconv.Atoi(input)

	fmt.Println(fmt.Sprintf("%d÷%d=%d...%d", a, b, a / b, a % b))
}

func (k02 K02) question3() {
	fmt.Print("一つ目の商品の値段は？")
	var input string
	_, _ = fmt.Scanln(&input)
	priceA, _ := strconv.Atoi(input)
	fmt.Print("個数は？")
	_, _ = fmt.Scanln(&input)
	amountA, _ := strconv.Atoi(input)

	fmt.Print("二つ目の商品の値段は？")
	_, _ = fmt.Scanln(&input)
	priceB, _ := strconv.Atoi(input)
	fmt.Print("個数は？")
	_, _ = fmt.Scanln(&input)
	amountB, _ := strconv.Atoi(input)

	fmt.Println(fmt.Sprintf("お支払いは税込み￥%fです", float32(priceA * amountA + priceB * amountB) * 1.08))
}

func (k02 K02) question4() {
	golem := Golem{300, 80, 50}

	fmt.Println(fmt.Sprintf("ゴーレム　（HP:%d　防御力：%d）", golem.Hp, golem.Defense))
	fmt.Println(fmt.Sprintf("HP：%d", golem.Hp))
	fmt.Print("今回の攻撃の値を入力してください＞")
	var input string
	_, _ = fmt.Scanln(&input)
	atk, _ := strconv.Atoi(input)

	if atk - golem.Defense < 0 {
		atk = 0
	} else {
		atk -= golem.Defense
	}
	fmt.Println(fmt.Sprintf("ダメージは%dです。", atk))
	golem.Hp -= atk
	fmt.Println(fmt.Sprintf("残りのHPは%dです。", golem.Hp))
}

func (k02 K02) execute(num int) {
	switch num {
	case 1:
		k02.question1()
	case 2:
		k02.question2()
	case 3:
		k02.question3()
	case 4:
		k02.question4()
	}
}

File: Nim/K02.nim
import os
from strutils import parseInt
import IExecutable, Golem

type K02* = ref object of IExecutable

method question1*(this: K02) =
    var seisuu = 3
    var jissuu: float32 = 2.6
    var moji = 'A'
    echo "変数seisuuの値は", seisuu
    echo "変数jissuuの値は", jissuu
    echo "変数mojiの値は", moji

method question2*(this: K02) =
    stdout.write "一つ目の整数は？"
    stdout.flushFile
    var a = parseInt(readLine(stdin))
    stdout.write "二つ目の整数は？"
    stdout.flushFile
    var b = parseInt(readLine(stdin))
    echo a, "÷", b, "=", a div b, "...", a mod b

method question3*(this: K02) =
    stdout.write "一つ目の商品の値段は？"
    stdout.flushFile
    var priceA = parseInt(readLine(stdin))
    stdout.write "個数は？"
    stdout.flushFile
    var amountA = parseInt(readLine(stdin))
    stdout.write "二つ目の商品の値段は？"
    stdout.flushFile
    var priceB = parseInt(readLine(stdin))
    stdout.write "個数は？"
    stdout.flushFile
    var amountB = parseInt(readLine(stdin))
    var total = (priceA * amountA + priceB * amountB).toFloat * 1.08
    echo "お支払いは税込み￥", total, "です"

method question4*(this: K02) =
    var golem = Golem(hp: 300, defense: 80, attack: 50)
    
    echo "ゴーレム　（HP：", golem.hp, "　防御力：", golem.defense, "）"
    echo "HP：", golem.hp
    stdout.write "今回の攻撃の値を入力してください＞"
    stdout.flushFile
    var damage = parseInt(readLine(stdin))
    damage -= golem.defense
    echo "ダメージは", damage, "です。"
    golem.hp -= damage
    echo "残りのHPは", golem.hp, "です。"